internal/cluster/distribution/eks/ekscluster/eks.go

```diff
@@ -31,10 +32,11 @@ import (
 
 // CreateClusterEKS describes Pipeline's Amazon EKS fields of a CreateCluster request
 type CreateClusterEKS struct {
-	Version      string               `json:"version,omitempty" yaml:"version,omitempty"`
-	NodePools    map[string]*NodePool `json:"nodePools,omitempty" yaml:"nodePools,omitempty"`
-	Vpc          *ClusterVPC          `json:"vpc,omitempty" yaml:"vpc,omitempty"`
-	RouteTableId string               `json:"routeTableId,omitempty" yaml:"routeTableId,omitempty"`
+	Version          string                  `json:"version,omitempty" yaml:"version,omitempty"`
+	EncryptionConfig []*eks.EncryptionConfig `json:"encryptionConfig,omitempty" yaml:"encryptionConfig,omitempty"`
```

Comment:
I'm not sure how I feel about using an EKS structure directly. Providing the same API is one thing, but this is a direct leak of the abstraction. I know transformation can be tedious...but it's kind of a coincidence that we expose the same API here. We have generated types, I'd rather use those and do a one time conversion somewhere. Then we can clearly separate our API from the underlying EKS API.

But as long as they are the same, I guess this is fine for the short-term.

Reply:
Then I'm going to typedef the AWS structure locally.

Reply:
(Or generate something somewhere if possible.)

Reply:
There is already a generated version in the API package. But I would only use it on the HTTP API surface.

Reply:
Should I duplicate the redefine in `internal/cluster/distribution/eks/ekscluster` and import that package into the `.../eksprovider/workflow` sources or should I duplicate the redefine in those 2 packages?
Not sure if architecturally we allow such relation between the two, but if a duplication can be avoided I would do so, because I'm having a hard time accepting conceptually there is no single place to define this.
(For example `Subnet` is identically duplicated in the two.)

Reply:
I implemented the duplicate redefinition approach [here](https://github.com/banzaicloud/pipeline/compare/6472650b045852246c96e9524c145a422a7049b6..f5bb49ddce4628811c995a6da8cf2d09409a1d0b).
It's easy to move to any other possible implementation choice from here.